bazel/cc_wrapper.py

```diff
@@ -1,128 +0,0 @@
-      elif "-stdlib=libc++" in envoy_cxxflags:
-        os.write(out_fd, "-l:libc++.a\n-l:libc++abi.a\n")
-
-
-# Is the arg a flag indicating that we're building for C++ (rather than C)?
-def is_cpp_flag(arg):
-  return arg in ["-static-libstdc++", "-stdlib=libc++", "-lstdc++", "-lc++"
-                ] or arg.startswith("-std=c++") or arg.startswith("-std=gnu++")
-
-
-def modify_driver_args(input_driver_flags):
-  # Detect if we're building for C++ or vanilla C.
-  if any(map(is_cpp_flag, input_driver_flags)):
-    # Append CXXFLAGS to all C++ targets (this is mostly for dependencies).
-    argv = shlex.split(envoy_cxxflags)
-  else:
-    # Append CFLAGS to all C targets (this is mostly for dependencies).
-    argv = shlex.split(envoy_cflags)
-
-  # Either:
-  # a) remove all occurrences of -lstdc++ (when statically linking against libstdc++),
-  # b) replace all occurrences of -lstdc++ with -lc++ (when linking against libc++).
-  if "-static-libstdc++" in input_driver_flags or "-stdlib=libc++" in envoy_cxxflags:
-    for arg in input_driver_flags:
-      if arg in ("-lstdc++", "-static-libstdc++"):
-        pass
-      elif arg.startswith("-Wl,@"):
-        # tempfile.mkstemp will write to the out-of-sandbox tempdir
-        # unless the user has explicitly set environment variables
-        # before starting Bazel. But here in $PWD is the Bazel sandbox,
-        # which will be deleted automatically after the compiler exits.
-        (flagfile_fd, flagfile_path) = tempfile.mkstemp(dir="./", suffix=".linker-params")
-        with closing_fd(flagfile_fd):
-          sanitize_flagfile(arg[len("-Wl,@"):], flagfile_fd)
-        argv.append("-Wl,@" + flagfile_path)
-      else:
-        argv.append(arg)
-  else:
-    argv += input_driver_flags
-
-  # This flags should after all libraries
-  if "-static-libstdc++" in input_driver_flags:
-    argv.append("-l:libstdc++.a")
-  if "-lstdc++" in input_driver_flags and "-stdlib=libc++" in envoy_cxxflags:
-    argv.append("-l:libc++.a")
-    argv.append("-l:libc++abi.a")
-
-  # Bazel will add -fuse-ld=gold in some cases, gcc/clang will take the last -fuse-ld argument,
-  # so whenever we see lld once, add it to the end.
-  if "-fuse-ld=lld" in argv:
-    argv.append("-fuse-ld=lld")
-
-  # Add compiler-specific options
-  if "clang" in compiler:
-    # This ensures that STL symbols are included.
-    # See https://github.com/envoyproxy/envoy/issues/1341
-    argv.append("-fno-limit-debug-info")
-    argv.append("-Wthread-safety")
-    argv.append("-Wgnu-conditional-omitted-operand")
-  elif "gcc" in compiler or "g++" in compiler:
-    # -Wmaybe-initialized is warning about many uses of absl::optional. Disable
-    # to prevent build breakage. This option does not exist in clang, so setting
-    # it in clang builds causes a build error because of unknown command line
-    # flag.
-    # See https://github.com/envoyproxy/envoy/issues/2987
-    argv.append("-Wno-maybe-uninitialized")
```

Comment:
In principle, this PR is really great, it will hugely simplify build. One downside is we lose an interception point to workaround Bazel issues, but I guess we can always introduce it again later. Beyond `libc++` handling, it looks like this line is something that we were using `cc_wrapper` for previously, what happens to this?

Reply:
In my TODO above:

> migrate compiler options to envoy_copts with select

will update for this and make it ready for review.